sgkit/io/vcfzarr_reader.py

```diff
@@ -221,39 +222,107 @@ def _concat_zarrs_optimized(
 
     first_zarr_group = zarr_groups[0]
 
-    with zarr.open_group(str(output)) as output_zarr:
+    # create the top-level group
+    zarr.open_group(str(output), mode="w")
```

Comment:
Is it intentional to open the zarr group twice, one with a context manager and one without?

Reply:
Good question. The reason I did this is because the dask computation doesn't actually use the `output_zarr` variable. So the sequence of operations is: 1. create group, 2. store (rechunked) arrays within the group (including attributes for those arrays, which is the goal of this PR), 3. copy unchanged arrays and top-level metadata.

Reply:
OK, sounds good. If this works I think we should try to get the logic down as some comments here. I can easily see someone coming along and "fixing" things. At least some comments saying "HERE BE DRAGONS!" or something.

Reply:
Good idea. I have added more comments along those lines.